src/shogun/base/SGObject.h

```diff
@@ -278,6 +282,119 @@ class CSGObject
 	 */
 	void build_gradient_parameter_dictionary(CMap<TParameter*, CSGObject*>* dict);
 
+	/** Sets a value for a parameter which is identified by a tag.
+	 *
+	 * @param _tag name and type information of parameter
+	 * @param value value of the parameter
+	 * @return current value of the parameter if it existed, otherwise
+	 * returns new value that is being set.
+	 */
+	template <typename T>
+	inline T set(Tag<T>& _tag, const T& value)
+	{
+		return recall_type<T>(_set(_tag, erase_type(value)));
+	}
+
+	/** Sets a value for a parameter which is identified by a name.
+	 *
+	 * @param name name of the parameter
+	 * @param value value of the parameter along with type information
+	 * @return current value of the parameter if it existed, otherwise
+	 * returns new value that is being set.
+	 */
+	template <typename T, typename U=void>
+	inline T set(const std::string& name, const T& value)
```

Comment:
As mentioned in the unit test comment, we need to `set`
One for *adding* elements to the map, called from constructor/`register_params` (just like SG_ADD).
And one for *setting* values of elements.
First one needs to be protected, i.e. from outside, users cannot add members